ltm/agent.py

```diff
@@ -459,7 +463,7 @@ def get_relevant_memories(self, user_message, cost_cb):
                 if r_mem.relevance > 0.6:
                     relevance_filtered_mems.append(r_mem)
 
-        llm_filtered_interactions = self.llm_memory_filter(relevance_filtered_mems, query_dict["queries"], cost_cb)
+        llm_filtered_interactions = self.llm_memory_filter(relevance_filtered_mems, query_dict["queries"], user_message, cost_cb)
```

Comment:
Wouldn't be better to call the evaluator after this line, rather than adding the user message as a parameter just for calling it inside? Here you already have `llm_filtered_interactions` and `relevance_filtered_mems`, so in principle you could call it right after performing the filtering.

Reply:
So `relevance_filtered_mems` are memory chunks rather than interactions. What is required is that we call `interactions_from_retrieved_memories` inside the function to get the interactions that are then filtered. And we actually store those and the filtered interactions in the evaluator. 

I can refactor this to be a little more single responsibility, but it will add more parameters to the function, not fewer.

Reply:
Understood. In that case, It's fine like it is I think.